src/plugins/ui_feature/selection_input.ts

```diff
@@ -185,7 +185,7 @@ export class SelectionInputPlugin extends UIPlugin implements StreamCallbacks<Se
       ...values.map((xc, i) => ({
         xc,
         id: (this.ranges.length + i + 1).toString(),
-        color: getNextColor(),
+        color: colors[this.ranges.length + i + (1 % colors.length)],
```

Comment:
There is a slight problem with this function; once this.range.length +i is bigger than the length of color, then you are accessing an index that does not exist.
try 
```suggestion
        color: colors[(this.ranges.length + i) % colors.length],
```

note that I removed +1 on purpose otherwise, you start accessing the color of index 1 and not 0

Reply:
Yep realised that issue later on, and +1 was there just for an increment but since " i " is going to increment anyway removing it seems fair, plus it also covers skipping of first color